docs/trestle-spec.md

```diff
@@ -0,0 +1,525 @@
+
+~~~
+.
+├── .git
+├── .trestle
+├── dist 
+│   └── catalogs
+│       └── mycatalog.json 
+└── catalogs
+    └── mycatalog
+        ├── catalog.json
+        ├── metadata
+        │   ├── metadata.json
+        │   ├── revision-history
+        │   │   ├── revision-history.json        
+        │   │   ├── 00000__revision-history.json
+        │   │   ├── 00001__revision-history.json
+        │   │   └── 00002__revision-history.json       
+        │   └── responsible-parties
+        │       ├── responsible-parties.json        
+        │       ├── creator__responsible-party.json
+        │       └── contact__responsible-party.json       
+        └── groups
+            ├── groups.json        
+            ├── 00000__group
+            │   ├── group.json
+            │   └── controls
+            │       ├── controls.json
+            │       ├── 00000__control.json
+            │       └── 00001__control.json
+            └── 00001__group
+                ├── group.json
+                └── controls
+                    ├── controls.json                
+                    ├── 00000__control.json
+                    ├── 00001__control.json
+                    └── 00002__control.json
+...
+~~~
+
+#### `trestle merge`
+
+The trestle merge command is the reversal of `trestle split`.
+
+#### `trestle build`
+
+This command merges all contents (files and directories) representing a specific model into a single OSCAL file located under `dist` folder. For example, `trestle merge catalog -i mycatalog` will traverse the `catalogs/mycatalog` directory and its children and combine all data into a OSCAL file that will be written to `dist/catalogs/mycatalog.json`. Note that the parts of catalog `mycatalog` can be written in either YAML/JSON/XML (e.g. based on the file extension), however, the output will be generated as YAML/JSON/XML as desired. Trestle will infer the content type from the file extension and create the model representation appropriately in memory and then output in the desired format. Trestle merge will also validate content as it assembles the files and make sure the contents are syntactically correct.
+
+#### `trestle add`
+
+This command allows users to add an OSCAL model to a subcomponent in source directory structure of the model. For example, `trestle add -e metadata.roles -i catalogs/mycatalog` will add the following property under the `metadata` property for a catalog that will be written to the appropriate file under `catalogs/mycatalog` directory:
+
+~~~
+"roles": [
+  {
+    "id": "REPLACE_ME",
+    "title": "REPLACE_ME"
+  }
+~~~
+
+Default values for each datatypes will be like below:
```

Comment:
Default values for mandatory datatypes will be like below. All UUID's will be populated by default whether or not they are mandatory.